WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -524,10 +524,15 @@ function(item, ev) {
 
         this._selectedItems.add(item);
 
-		if (item._setSelected(true)) {
-            this._updateHistory(item, true, isShowInHistory);
-			this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
-		}
+        if (item._setSelected(true)) {
+            if ((item._text === ZaMsg.NAD_Tab_HSM && typeof ZaHSM !== 'undefined' && ZaHSM.checkstoragefeaturenabled()) ||
```

Comment:
Does this work?
```suggestion
            if ((item._text === ZaMsg.NAD_Tab_HSM && ZaHSM && ZaHSM.checkstoragefeaturenabled()) ||
```

Reply:
okay, it is harmless anyway. I thought there might be a case that ZaHSM is not undefined but null. If not, the current code to compare with `undefined` is fine.